src/mpi-base.jl

```diff
@@ -525,6 +525,62 @@ function Reduce{T}(object::T, op::Op, root::Integer, comm::Comm)
     isroot ? recvbuf[1] : nothing
 end
 
+# T and T2 are different for MPI_IN_PLACE as the sendbuf
+function Allreduce{T}(sendbuf::MPIBuffertype{T}, recvbuf::MPIBuffertype{T},
+                      count::Integer, op::Op, comm::Comm)
+    flag = zero(Cint)
+    ccall(MPI_ALLREDUCE, Void, (Ptr{T}, Ptr{T}, Ptr{Cint}, Ptr{Cint}, Ptr{Cint},
+          Ptr{Cint}, Ptr{Cint}), sendbuf, recvbuf, &Int32(count), &mpitype(T), 
+          &op.val, &comm.val, &flag)
+
+    if flag != 0
+      println(STDERR, "Warning: MPI_Allreduce exited with status ", flag)
```

Comment:
Is there a consistent policy on handling return codes?  Most of the functions don't check the `ierr` flag, but I'm not sure if that is an explicit decision to ignore them or just a feature that hasn't been supported yet.